ptf/sai_base_test.py

```diff
@@ -0,0 +1,945 @@
+            debug (bool): enables debug option
+        Return:
+            dict: switch_resources dictionary with available resources
+        """
+
+        switch_resources = sai_thrift_get_switch_attribute(
+            self.client,
+            available_ipv4_route_entry=True,
+            available_ipv6_route_entry=True,
+            available_ipv4_nexthop_entry=True,
+            available_ipv6_nexthop_entry=True,
+            available_ipv4_neighbor_entry=True,
+            available_ipv6_neighbor_entry=True,
+            available_next_hop_group_entry=True,
+            available_next_hop_group_member_entry=True,
+            available_fdb_entry=True,
+            available_ipmc_entry=True,
+            available_snat_entry=True,
+            available_dnat_entry=True,
+            available_double_nat_entry=True,
+            number_of_ecmp_groups=True,
+            ecmp_members=True)
+
+        if debug:
+            self.printNumberOfAvaiableResources(switch_resources)
+
+        return switch_resources
+
+    def verifyNumberOfAvaiableResources(self, init_resources, debug=False):
+        """
+        Verify number of available resources
+
+        Args:
+            init_resources (dict): a dictionary with initial resources numbers
+            debug (bool): enable debug option
+
+        Returns:
+            bool: True if the numbers of resources are the same as before tests
+        """
+
+        available_resources = sai_thrift_get_switch_attribute(
+            self.client,
+            available_ipv4_route_entry=True,
+            available_ipv6_route_entry=True,
+            available_ipv4_nexthop_entry=True,
+            available_ipv6_nexthop_entry=True,
+            available_ipv4_neighbor_entry=True,
+            available_ipv6_neighbor_entry=True,
+            available_next_hop_group_entry=True,
+            available_next_hop_group_member_entry=True,
+            available_fdb_entry=True,
+            available_ipmc_entry=True,
+            available_snat_entry=True,
+            available_dnat_entry=True,
+            available_double_nat_entry=True,
+            number_of_ecmp_groups=True,
+            ecmp_members=True)
+
+        for key, value in available_resources.items():
+            if value != init_resources[key]:
+                if debug:
+                    print("Number of %s incorrect!" % key)
+                return False
+
+        return True
+
+    @staticmethod
+    def status():
+        """
+        Returns the last operation status.
+
+        Returns:
+            int: sai call result
+        """
+        return adapter.status
+
+    @staticmethod
+    def saiWaitFdbAge(timeout):
+        """
+        Wait for fdb entry to ageout
+
+        Args:
+            timeout (int): Timeout value in seconds
+        """
+        print("Waiting for fdb entry to age")
+        aging_interval_buffer = 10
+        time.sleep(timeout + aging_interval_buffer)
+
+
+class SaiHelper(SaiHelperBase):
```

Comment:
Does it make sense to rename this class since it sets up a specific configuration that all tests may not be able to use.

Reply:
Do you mean this class should not be named "SaiHelper"? Actually this class "helps" to set designed configuration which we use in most of the tests. This is even more common than using SaiHelperBace as a parent class.

Reply:
Most tests which are being upstreamed use this as the base class and use the configuration defined in this class. New tests may choose to use SaiHelperBase instead to define their own configuration.